slither/detectors/statements/unprotected_upgradeable.py

```diff
@@ -21,12 +21,22 @@ def _can_be_destroyed(contract: Contract) -> List[Function]:
                 break
     return targets
 
+def _has_initializing_protection(functions: List[Function]) -> bool:
+    # Detects "initializer" constructor modifiers and "_disableInitializers()" constructor internal calls
+    # https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#initializing_the_implementation_contract
 
-def _has_initializer_modifier(functions: List[Function]) -> bool:
     for f in functions:
         for m in f.modifiers:
             if m.name == "initializer":
                 return True
+        # filtering out SolidityFunction from the internal calls as we don't need to match against those names
+        internal_func_calls = [c for c in f.all_internal_calls() if not isinstance(c, SolidityFunction)]
```

Comment:
This loops over the list twice. I think it'd be better to remove the list comprehension and iterate once over `f.all_internal_calls()`

Reply:
Good suggestion!